src/types.ml

```diff
@@ -50,23 +50,26 @@ let rec binds = function
   | PMatch t -> []
   | PFunc(_, args) | PForm(_, args) | PTuple args -> List.concat (List.map binds args)
 
-(* Channel options / Bullet notation
-type channel_options = { authentic: bool; secret: bool }
-*)
+(* Channel options / Bullet notation *)
+type channel_option =
+  Public
+  | Auth
+  | Conf
+  | AuthConf
 
 (* Global types: p -> q *)
 type global_type =
-    Send of principal * principal * ident * term * global_type
-  | Branch of principal * principal * term * (pattern * global_type) list
+    Send of principal * principal * channel_option * ident * term * global_type
+  | Branch of principal * principal * channel_option * term * (pattern * global_type) list
   | Compute of principal * let_bind * global_type
   | DefGlobal of ident * (ident * principal) list * global_type * global_type
   | CallGlobal of ident * term list
   | GlobalEnd
 
 (* Local Type *)
 type local_type =
-    LSend of principal * term * local_type
-  | LRecv of principal * pattern * term * local_type
+    LSend of ident * channel_option * term * local_type
+  | LRecv of ident * channel_option * pattern * term * local_type
```

Comment:
Why do we have `ident` here?

Reply:
I have changed it to be a joined name of both the sender and the receiver.
This means that a message between Alice and Bob should end up having the ident "AliceBob". It should be sorted, meaning it always ends up being "AliceBob" even if Bob is the sender.